File: .tmp_c4000/batch_04_go/3457_eat_pizzas.go
// LeetCode 3457 - Eat Pizzas!
// https://leetcode.com/problems/eat-pizzas/

import "sort"

func maxWeight(pizzas []int) int64 {
	sort.Ints(pizzas)
	n := len(pizzas)
	days := n / 4
	var ans int64
	oddDays := (days + 1) / 2
	evenDays := days / 2
	idx := n - 1
	for i := 0; i < oddDays; i++ {
		ans += int64(pizzas[idx])
		idx--
	}
	for i := 0; i < evenDays; i++ {
		idx-- // skip one
		ans += int64(pizzas[idx])
		idx--
	}
	return ans
}

File: .tmp_py4013/batch_13_dump/3457_eat_pizzas.js
// LeetCode 3457 - Eat Pizzas!
// https://leetcode.com/problems/eat-pizzas/

var maxWeight = function(pizzas) {
    pizzas = pizzas.slice().sort((a, b) => a - b);
    const n = pizzas.length;
    const days = Math.floor(n / 4);
    let ans = 0;
    const oddDays = Math.floor((days + 1) / 2);
    const evenDays = Math.floor(days / 2);
    let idx = n - 1;
    for (let i = 0; i < oddDays; i++) {
        ans += pizzas[idx];
        idx--;
    }
    for (let i = 0; i < evenDays; i++) {
        idx--;
        ans += pizzas[idx];
        idx--;
    }
    return ans;
};
